### PR TITLE
Prepare jQuery upload integration by updating upload class #2641

### PR DESCRIPTION
**Pull request by xillibit**
This change is needed to jQuery upload to handle multiples files at same times, i have tested on blue eagle template and no issues appears.

Instead of make a big PR for jQuery uploader i have separated a bit my works to reduce a bit the PR about jQuery uploader.

### CODE REVIEW

File: libraries/kunena/upload/upload.php [hunk 1]
```diff
@@ -4,7 +4,7 @@
  * @package     Kunena.Framework
  * @subpackage  Upload
  *
- * @copyright   (C) 2008 - 2015 Kunena Team. All rights reserved.
+ * @copyright   (C) 2008 - 2014 Kunena Team. All rights reserved.
```
Comment (810) [inline libraries/kunena/upload/upload.php]: 2015

Comment (xillibit) [inline libraries/kunena/upload/upload.php]: Updated

**Comment by 810** [comment]
- [x] Multiple images (success)
- [ ] Multiple files (failed)  - Cannot save attachment: File '/' doesn't exist!

**Comment by xillibit** [comment]
I have updated this PR, try now it should be better

**Comment by rich20** [comment]
Blue eagle

- [x] single image (success)
- [x] single file (success)
- [x] Multiple images (success)
- [x] Multiple files (success)

Crypsis

- [x] single image (success)
- [x] single file (success)
- [x] Multiple images (success)
- [x] Multiple files (success)

**Comment by 810** [comment]
Blue eagle
- [x] single image (success)
- [x] single file (success)
- [x] Multiple images (success)
- [x] Multiple files (success)
- [x] Multiple files - correct error message on reached max upload (success)

Crypsis
- [x] single image (success)
- [x] single file (success)
- [x] Multiple images (success)
- [x ] Multiple files (success)
- [x] Multiple files - correct error message on reached max upload (success)

File: components/com_kunena/site/controllers/topic.php [hunk 2]
```diff
@@ -268,12 +268,17 @@ public function post() {
...
+		// Legacy way to upload new attachments
+		foreach ($_FILES as $key => $file)
+		{
 			$intkey = 0;
 			if (preg_match('/\D*(\d+)/', $key, $matches))
-				$intkey = (int)$matches[1];
-			if ($file['error'] != UPLOAD_ERR_NO_FILE) $message->uploadAttachment($intkey, $key, $this->catid);
+			$intkey = (int)$matches[1];
+
+			if ($file['error'] != 4)
```
Comment (coder4life) [inline components/com_kunena/site/controllers/topic.php]: Why did you change the constant?

Comment (xillibit) [inline components/com_kunena/site/controllers/topic.php]: because it works with integer and not a constant

File: libraries/kunena/upload/upload.php [hunk 3]
```diff
@@ -285,75 +299,115 @@ public function ajaxUpload(array $options)
...
-				{
-					throw $exception;
-				}
-
-				$in = fopen($_FILES['file']['tmp_name'], 'rb');
-			}
-			else
-			{
-				// Multi-part upload.
-				$in = fopen('php://input', 'rb');
```
Comment (coder4life) [inline libraries/kunena/upload/upload.php]: Why is the stream php://input removed.  We been trying to use streams, my suspicion though it is a behavior of php://input, I know there were a few problems.

File: components/com_kunena/site/controllers/topic.php (hunk 2 above)
Comment (coder4life) [inline components/com_kunena/site/controllers/topic.php]: Yes but I thought the constant is representative of an INTEGER value

**Comment by xillibit** [comment]
I'am closing this PR for now because i found a way to avoid to modify that class
